src/rebar3_sbom_cyclonedx.erl

```diff
@@ -1,22 +1,36 @@
 -module(rebar3_sbom_cyclonedx).
 
--export([bom/1, bom/2, uuid/0]).
+-export([bom/2, bom/3, uuid/0]).
 
-bom(Components) ->
-    bom(Components, uuid()).
+-define(APP, "rebar3_sbom").
+-define(COMPONENT_FIELDS, [name, version, author, description, licenses, purl, sha256]).
 
-bom(Components, Serial) ->
-    Bom = {bom, [{serialNumber, Serial}, {xmlns, "http://cyclonedx.org/schema/bom/1.1"}], [
-        {components, [], [component(Component) || Component <- Components, Component /= undefined]}
-    ]},
+-include_lib("xmerl/include/xmerl.hrl").
+
+bom(File, Components) ->
+    bom(File, Components, uuid()).
+
+bom(File, Components, Serial) ->
+    Bom = {bom, [{version, [get_version(File)]}, {serialNumber, Serial}, {xmlns, "http://cyclonedx.org/schema/bom/1.1"}], [
+        {metadata, metadata()},
+        {components, [], [component(Component) || Component <- Components, Component /= undefined]},
+        {dependencies, [], [dependency(Component) || Component <- Components, Components /= undefined]}]},
     xmerl:export_simple([Bom], xmerl_xml).
 
+metadata() ->
+    [{timestamp, [calendar:system_time_to_rfc3339(erlang:system_time(second))]},
+     {tools, [{tool, [?APP]}]},
+     {licenses, [{license, [], [{id, [], [["BSD-3-Clause"]]}]}]}
```

Comment:
This metadata element is optional. Since we are not really in a position to decide under what license the user intends to release the SBoM I think we should omit this element, or else offer a command line argument to choose a value.

Reply:
I removed the licenses line, but I think maybe `licenses` just refers to the license of rebar3_sbom itself

Reply:
Sorry, I have not tried processing the out. What I need is to output a JSON bom file with 1.4 specs, and the first step for me is to upgrade the XML to 1.4. If compatibility is essential, we can add a version select option in the command line options.

Reply:
>Sorry, I have not tried processing the out.

I will try and give it a spin myself later today, and see if my old(ish) DependencyTrack instance chokes on the 1.4 format...